### PR TITLE
feat(ST-66): implement delete resource endpoint

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Updated the file-upload file to include delete from S3 method
- Updated resources  service and repository for adding deletion method
- Included endpoint in controller for deleting resource

## Associated Ticket Link(s)
- https://trello.com/c/9U4CJWJu

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/file-uploads/file-uploads.service.ts [hunk 1]
```diff
@@ -34,4 +36,15 @@ export class FileUploadsService {
     });
     return response;
   }
+
+  async deleteFromS3(key: string) {
+    console.log("key", key);
```
Comment (ishmam-iqbal-sazim) [inline src/file-uploads/file-uploads.service.ts]: Let's remove unnecessary console logs

File: src/resources/resources.service.ts [hunk 2]
```diff
@@ -46,4 +46,16 @@ export class ResourcesService {
 
     return this.resourcesRepository.find({ classroom: classroom });
   }
+
+  async deleteResource(resourceId: number) {
+    const resource = await this.resourcesRepository.findOneOrFail(resourceId);
+
+    const fileKey = resource.fileUrl.split("project-dev-bucket/")[1];
+
+    await this.fileUploadsService.deleteFromS3(fileKey);
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.service.ts:55]: Don't we need any checks here or inside the method for fail cases?

Comment (ghost) [reply src/resources/resources.service.ts:55]: Yes bhaiya, I will use try and catch block on the fileUpload.service layer

**Review by ragib-hossain-sazim — APPROVED** [review]

File: src/resources/resources.controller.ts [hunk 3]
```diff
@@ -15,11 +24,16 @@ export class ResourcesController {
     @Body("title") title: string,
     @Body("description") description: string,
   ) {
-    return this.classworksService.uploadResource(file, classroomId, title, description);
+    return this.resourcesService.uploadResource(file, classroomId, title, description);
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.controller.ts:27]:
```suggestion
    return this.resourcesService.uploadOne(file, classroomId, title, description);
```
[NIT]

File: src/resources/resources.controller.ts [hunk 4]
```diff
@@ -15,11 +24,16 @@ export class ResourcesController {
...
 
   @Get(":classroomId/resources")
   getClassroomResources(@Param("classroomId") classroomId: number) {
-    return this.classworksService.getClassroomResources(classroomId);
+    return this.resourcesService.getClassroomResources(classroomId);
+  }
+
+  @Delete(":classroomId/resources/:resourceId")
+  async deleteResource(@Param("resourceId") resourceId: number) {
+    await this.resourcesService.deleteResource(resourceId);
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.controller.ts:37]:
```suggestion
    await this.resourcesService.deleteOne(resourceId);
```
[NIT]

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Pull request merged**